src/mudlet.cpp

```diff
@@ -691,16 +694,92 @@ mudlet::mudlet()
 //    });
 }
 
-QSettings* mudlet::getQSettings()
+static QString findExecutableDir() {
+    // Linux AppImage support
+    QProcessEnvironment systemEnvironment = QProcessEnvironment::systemEnvironment();
+    if (systemEnvironment.contains(qsl("APPIMAGE"))) {
+        QString appimgPath = systemEnvironment.value(qsl("APPIMAGE"), QString());
+        return QFileInfo(appimgPath).dir().path();
+    }
+    return QCoreApplication::applicationDirPath();
+}
+
+static QString readMarkerFile(const QString& path) {
+    QString line;
+    QFile file(path);
+    file.open(QIODevice::ReadOnly);
+    QTextStream(&file).readLineInto(&line);
+    file.close();
+    return line;
+}
+
+static bool validateConfDir(QString& path) {
+    if (path.size() == 0) {
+        qWarning("WARN: portable data path not specified");
+        return false;
+    }
+    QFileInfo pathInfo(path);
+    if (pathInfo.isFile()) {
+        qWarning("WARN: specified portable data path is an existing file: %s", qPrintable(path));
+        return false;
+    }
+    QFileInfo parentInfo(pathInfo.dir().path());
+    if (!parentInfo.isDir()) {
+        qWarning("WARN: parent directory of specified portable data path doesn't exist: %s", qPrintable(parentInfo.filePath()));
+        return false;
+    }
+    return true;
+}
+
+static void migrateConfig(QSettings& settings) {
+    if (settings.contains("pos")) {
```

Comment:
It might have been picked up elsewhere but please don't use raw string literals in the C++ code:

* Either - if they are only used internally and not visible to the end-user or are output to the user from the Lua sub-system:  wrap them in `qsl(...)` (actually our macro for the more to type `QStringLiteral(...)`). This case applies here!
* Or - if they are presented to the end-user in the UI - wrap them in `tr(...)` (actually `QObject::tr(...)`) or for use in source files/classes not including the `QObject` or `Q_DECLARE_TR_FUNCTIONS` in their header files or - like `main.cpp` - not having a header file - in `QApplication::translate(...)`.